src/lib/dns.coffee

```diff
@@ -165,14 +142,12 @@ module.exports = (dnschain) ->
                                 return @sendErr res, NAME_RCODE.SERVFAIL, cb
 
             else if S(q.name).endsWith '.dns'
-                # TODO: right now we're doing a catch-all and pretending they asked
-                #       for namecoin.dns...
                 res.answer.push gIP2type(q.name,ttl,QTYPE_NAME[q.type])(gConf.get 'dns:externalIP')
                 @log.debug gLineInfo('cb|.dns'), {q:q, answer:res.answer}
                 @sendRes res, cb
             else
                 @log.debug gLineInfo("deferring request"), {q:q}
-                @oldDNSLookup req, (packet, code) =>
+                @dnschain.cache.resolveOldDNS req, (code, packet) =>
                     _.assign res, _.pick packet, ['edns_version', 'edns_options',
                         'edns', 'answer', 'authority', 'additional']
```

Comment:
@WeMeetAgain: In your implementation of `resolveOldDNS`, can you *just* cache the items in this array instead of the entire packet itself?

If you can do that, then I think we might actually be able to use Redis to do caching of DNS requests. Remember that responses must be tailored to each client (they've got different IPs, after all).